desc/optimize/_constraint_wrappers.py

```diff
@@ -768,6 +802,36 @@ def hess(self, x):
         J = self.jac_scaled(x)
         return J.T @ J
 
+    def vjp_scaled(self, v, x):
+        """Compute vector-Jacobian product of the objective function.
+
+        Uses the scaled form of the objective.
+
+        Parameters
+        ----------
+        v : ndarray
+            Vector to left-multiply the Jacobian by.
+        x : ndarray
+            Optimization variables.
+
+        """
+        raise NotImplementedError
```

Comment:
couldn't this just call the jac_scaled and multiply the two vectors?

Reply:
Technically yes but that's the expensive way that this is trying to avoid, so it seemed better to make the user do that explicitly